connectors/src/connectors/zendesk/temporal/workflows.ts

```diff
@@ -388,6 +394,50 @@ export async function zendeskCategorySyncWorkflow({
   }
 }
 
+/**
+ * Garbage collection workflow for Zendesk.
+ *
+ * This workflow is responsible for deleting the following (in this order):
+ * - Outdated tickets.
+ * - Articles that cannot be found anymore in the Zendesk API.
+ * - Categories that have no article anymore.
+ * - Brands that have no permission on tickets and Help Center anymore.
+ */
+export async function zendeskGarbageCollectionWorkflow({
+  connectorId,
+}: {
+  connectorId: ModelId;
+}) {
+  // deleting the outdated tickets
+  let ticketIds = await getNextOldTicketBatchActivity(connectorId);
+  while (ticketIds.length > 0) {
+    await deleteTicketBatchActivity(connectorId, ticketIds);
+    ticketIds = await getNextOldTicketBatchActivity(connectorId);
+  }
+
+  // deleting the articles that cannot be found anymore in the Zendesk API
+  const brandIds =
+    await getZendeskHelpCenterReadAllowedBrandIdsActivity(connectorId);
+  for (const brandId of brandIds) {
+    let cursor = null;
+
+    do {
+      const { articleIds, cursor: newCursor } =
```

Comment:
similarly, I think these 2 activities could be merged to avoid storing data in workflow state (this one may be more important since we'll need to go over every single article every day)